engine/collection/compliance/core.go

```diff
@@ -200,21 +204,33 @@ func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.Clus
 func (c *Core) processBlockAndDescendants(proposal *messages.ClusterBlockProposal) error {
 	blockID := proposal.Header.ID()
 
+	log := c.log.With().
+		Str("block_id", blockID.String()).
+		Uint64("block_height", proposal.Header.Height).
+		Uint64("block_view", proposal.Header.View).
+		Logger()
+
 	// process block itself
 	err := c.processBlockProposal(proposal)
-	// child is outdated by the time we started processing it
-	// => node was probably behind and is catching up. Log as warning
-	if engine.IsOutdatedInputError(err) {
-		c.log.Info().Msg("dropped processing of abandoned fork; this might be an indicator that the node is slightly behind")
-		return nil
-	}
-	// the block is invalid; log as error as we desire honest participation
-	// ToDo: potential slashing
-	if engine.IsInvalidInputError(err) {
-		c.log.Warn().Err(err).Msg("received invalid block from other node (potential slashing evidence?)")
-		return nil
-	}
 	if err != nil {
+		if engine.IsOutdatedInputError(err) {
+			// child is outdated by the time we started processing it
+			// => node was probably behind and is catching up. Log as warning
+			log.Info().Msg("dropped processing of abandoned fork; this might be an indicator that the node is slightly behind")
+			return nil
+		}
+		if engine.IsInvalidInputError(err) {
```

Comment:
:warning: At this point we need to notify vote aggregator that there is an invalid block to potentially slash byzantine voters. We could directly notify `VoteAggregator` or use a pub-sub mechanism(requires extending `Consumer` I suppose)

Reply:
👍 I will notify directly for now. We also call `AddVote` directly and they seem to be similar cost.